### ISSUE TITLE
Can't do `find_or_create_by(!)` on associations

### ISSUE THREAD

**Issue by cheerfulstoic** [neo4jrb/neo4j#1240]
Model:

```ruby
class Neo4jDevice
  include Neo4j::ActiveNode

  property :hostname, type: String, index: :exact
  property :iosversion, type: String
  property :device_type, type: String
  # more properties

  has_many :out, :neo4j_interfaces, type: :connected_to
end

class Neo4jInterface 
  include Neo4j::ActiveNode

  property :ifname, type: String, index: :exact
  property :ifstatus, type: String
  property :type, type: String
  #... more properties

  has_one :in, :neo4j_device, type: :connected_to
end
```

Code:
```ruby
device = Neo4jDevice.find_or_create_by!(hostname: "host")
interface = device.neo4j_interfaces.find_or_create_by!(ifname: "interface 0")
```

Creates the two objects, but not the association.  This works, though:

```ruby
device.neo4j_interfaces << (interface = Neo4jInterface.find_or_create_by!(ifname: "interface 0”))
```

**Comment by cheerfulstoic** [comment]
Reported by @sleeptillseven on Gitter

**Comment by efatsi** [comment]
Some more information here.

I've found that a `find_or_create_by` on a relationship works, and successfully creates the association (contrary to your report @cheerfulstoic), however, I've found another issue which is that if you try `find_or_create_by` on a relationship, the `find_by` that occurs appears to be unscoped to the parent node. eg:

Model:
```ruby
class Neo4jDevice
  include Neo4j::ActiveNode

  has_many :out, :neo4j_interfaces, type: :connected_to
end

class Neo4jInterface 
  include Neo4j::ActiveNode

  property :ifname, type: String, index: :exact

  has_one :in, :neo4j_device, type: :connected_to
end
```

Code:

```ruby
device_1    = Neo4jDevice.create
interface_1 = device_1.neo4j_interfaces.find_or_create_by(ifname: "Test")

device_2    = Neo4jDevice.create
interface_2 = device_2.neo4j_interfaces.find_or_create_by(ifname: "Test")

Neo4jInterface.count                                   #=> 1
device_1.neo4j_interfaces == [interface_1]             #=> true
device_2.neo4j_interfaces == [interface_2]             #=> true 
device_1.neo4j_interfaces == device_2.neo4j_interfaces #=> true
```

**Comment by cheerfulstoic** [comment]
Thanks for pointing this out!  It looks like this was expected behavior, but I really, really don't think that it should be.  Since it's a breaking change I just released it as `8.0.0.rc.4` so look for it there
